OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs

```diff
@@ -57,17 +58,23 @@ public class GrantConditionOnDeployInfo : PausableConditionalTraitInfo
 		[VoiceReference]
 		public readonly string Voice = "Action";
 
+		[Desc("Can this actor be ordered to move when deployed? [Always, ForceOnly, Never]")]
+		public readonly UndeployOnMoveType UndeployOnMove = UndeployOnMoveType.Always;
```

Comment:
The default behavior for `GrantConditionOnDeploy` is to have nothing to do with movement, so the default here really should be `UndeployOnMoveType.Never` to match. Mods must explicitly opt-in to having the deploy affect movement, so it it logical for them to opt in to having movement affect the deploy at the same time.

This otherwise breaks usecases that have nothing to do with movement - e.g. deploy to activate gap generation or cloak.